inst/include/dust/filter_tools.hpp

```diff
@@ -59,16 +59,55 @@ real_type scale_log_weights(typename std::vector<real_type>::iterator w, size_t
       max_w = std::max(max_w, *wi);
     }
   }
+
+  if (max_w == -std::numeric_limits<real_type>::infinity()) {
+    return max_w;
+  }
+
   real_type tot = 0.0;
   wi = w;
   for (size_t i = 0; i < n; ++i, ++wi) {
     *wi = std::exp(*wi - max_w);
     tot += *wi;
   }
+
   return std::log(tot / n) + max_w;
 }
 
 
+template <typename real_type>
+bool early_exit(const std::vector<real_type>& log_likelihood,
+                const std::vector<real_type>& min) {
+  for (auto x : log_likelihood) {
```

Comment:
Should this have a `&` to not copy the vector? I can never remember for these loops

Reply:
Here `x` is an atomic type (`real_type`, i.e., `double`/`float`) so it makes no difference and I think is more typical not to add a reference